src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -17,7 +17,7 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
     // This is a type symbol associated with a type definition in source code.
     // That is, for a generic type C<T> this is the instance type C<T>.  
-    internal sealed partial class SourceNamedTypeSymbol : SourceMemberContainerTypeSymbol, IAttributeTargetSymbol
```

Comment:
>sealed [](start = 13, length = 6)

Is this change necessary? (Ideally `Symbol` classes should be `sealed` or `abstract`.)

Reply:
I'm adding `InjectedNamespaceSymbol : SourceNamedTypeSymbol`.
I could pull a base type to restore sealed/abstract.
Another option is to merge `InjectedNamespaceSymbol` and `SourceNamedTypeSymbol` by adding a flag to `SourceNamedTypeSymbol`.

Reply:
`InjectedNamespaceSymbol` is derived from `SourceNamespaceSymbol` rather than this class.

And for `InjectedNamespaceSymbol`, if the only change from the base class is `IsImplicitlyDeclared`, then adding a flag to the base class as you suggested seems reasonable.

---
In reply to: [214401103](https://github.com/dotnet/roslyn/pull/29180#discussion_r214401103) [](ancestors = 214401103)